src/v/cluster/metrics_reporter.cc

```diff
@@ -0,0 +1,434 @@
+      force_refresh::no,
+      config::shard_local_cfg().metrics_reporter_report_interval()
+        + ss::lowres_clock::now());
+
+    if (!report) {
+        co_return result<metrics_snapshot>(report.error());
+    }
+    metrics_map.reserve(report.value().node_states.size());
+
+    for (auto& ns : report.value().node_states) {
+        auto [it, _] = metrics_map.emplace(ns.id, node_metrics{.id = ns.id});
+        auto& metrics = it->second;
+        metrics.is_alive = (bool)ns.is_alive;
+
+        auto broker = _members_table.local().get_broker(ns.id);
+        if (!broker) {
+            continue;
+        }
+
+        metrics.cpu_count = broker.value()->properties().cores;
+    }
+
+    for (auto& report : report.value().node_reports) {
+        auto it = metrics_map.find(report.id);
+        if (it == metrics_map.end()) {
+            auto [eit, _] = metrics_map.emplace(
+              report.id, node_metrics{.id = report.id});
+            it = eit;
+        }
+        auto& metrics = it->second;
+
+        metrics.version = report.redpanda_version;
+        metrics.disks.reserve(report.disk_space.size());
+        std::transform(
+          report.disk_space.begin(),
+          report.disk_space.end(),
+          std::back_inserter(metrics.disks),
+          [](const cluster::node_disk_space& nds) {
+              return node_disk_space{.free = nds.free, .total = nds.total};
+          });
+
+        metrics.uptime_ms = report.uptime / 1ms;
+    }
+    auto& topics = _topics.local().topics_map();
+    snapshot.topic_count = 0;
+    snapshot.partition_count = 0;
+    for (const auto& [tp_ns, md] : topics) {
+        // do not include internal topics
+        if (
+          tp_ns.ns == model::redpanda_ns
+          || tp_ns.ns == model::kafka_internal_namespace
+          || !md.is_topic_replicable()) {
+            continue;
+        }
+
+        snapshot.topic_count++;
+        snapshot.partition_count += md.get_configuration().cfg.partition_count;
+    }
+
+    snapshot.nodes.reserve(metrics_map.size());
+    for (auto& [_, m] : metrics_map) {
+        snapshot.nodes.push_back(std::move(m));
+    }
+
+    co_return snapshot;
+}
+
+ss::sstring format_uuid(const std::array<uint8_t, 16> buffer) {
+    std::stringstream ss;
+
+    for (size_t i = 0; i < buffer.size(); ++i) {
+        if (i == 4 || i == 6 || i == 8 || i == 10) {
+            ss << "-";
+        }
+        fmt::print(ss, "{:02x}", buffer.at(i));
+    }
+    return ss.str();
+}
+
+ss::future<> metrics_reporter::try_initialize_cluster_info() {
+    // already initialized, do nothing
+    if (_cluster_uuid != "") {
+        co_return;
+    }
+
+    storage::log_reader_config reader_cfg(
+      model::offset(0), model::offset(2), ss::default_priority_class());
+    reader_cfg.type_filter = model::record_batch_type::raft_configuration;
+    auto reader = co_await _raft0->make_reader(reader_cfg);
+
+    auto batches = co_await model::consume_reader_to_memory(
```

Comment:
Calling with `no_timeout`, is it abortable some other way if this is running during shutdown?

Reply:
it is not, but this is going to be really fast since there are only two batches, and they are very small.